llama_index/chat_engine/types.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List, Tuple
+
+from llama_index.response.schema import RESPONSE_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class BaseChatEngine(ABC):
```

Comment:
out of curisoity, what's the difference between a BaseChatEngine and a BaseQueryEngine? The high-level interface seems pretty similar

Reply:
One is stateful, one is not. In terms of function signature very similar.